qt/python/mantidqt/widgets/codeeditor/tab_widget/codeeditor_tab_view.py

```diff
@@ -0,0 +1,145 @@
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+from __future__ import absolute_import
+
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QColor
+from qtpy.QtWidgets import QAction, QHBoxLayout, QMenu, QPushButton, QSizePolicy, QTabWidget, QWidget
+
+from mantidqt.icons import get_icon
+from mantidqt.utils.qt import add_actions, create_action
+from mantidqt.widgets.codeeditor.tab_widget.codeeditor_tab_presenter import CodeEditorTabPresenter
+
+
+class CodeEditorTabWidget(QTabWidget):
+    ABORT_BUTTON_OBJECT_NAME = "abort-button"
+    NEW_EDITOR_PLUS_BTN_OBJECT_NAME = "plus-button"
+    OPTIONS_BUTTON_OBJECT_NAME = "options-button"
+    RUN_BUTTON_OBJECT_NAME = "run-button"
+    SHOW_IN_EXPLORER_ACTION_OBJECT_NAME = "show-in-explorer-action"
+
+    def __init__(self, parent=None, presenter=None):
+        self.presenter = presenter if presenter else CodeEditorTabPresenter(self)
+        super(CodeEditorTabWidget, self).__init__(parent)
+
+        self.setAttribute(Qt.WA_DeleteOnClose, True)
+        self.setContextMenuPolicy(Qt.ActionsContextMenu)
+        self.setMovable(True)
+        self.setTabsClosable(True)
+        self.setDocumentMode(True)
+
+        self.last_tab_clicked = 0
+
+        self.setup_tabs_context_menu(parent)
+        self.setup_options_actions(parent)
+
+        # create a button to add new tabs
+        plus_button = QPushButton(self)
+        plus_button.setObjectName(self.NEW_EDITOR_PLUS_BTN_OBJECT_NAME)
+        plus_button.clicked.connect(parent.plus_button_clicked)
+        plus_button.setIcon(get_icon("fa.plus"))
+        self.setCornerWidget(plus_button, Qt.TopLeftCorner)
+
+    def setup_tabs_context_menu(self, parent):
+        """
+        Setup the actions for the context menu (right click). These are handled by the presenter
+        """
+        self.tabBarClicked.connect(self.tab_was_clicked)
+
+        show_in_explorer = QAction("Show in Explorer", self)
+        show_in_explorer.setObjectName(self.SHOW_IN_EXPLORER_ACTION_OBJECT_NAME)
+        show_in_explorer.triggered.connect(self.presenter.action_show_in_explorer)
+
+        separator = QAction(self)
+        separator.setSeparator(True)
+
+        self.addAction(show_in_explorer)
+
+    def setup_options_actions(self, parent):
+        """
+        Setup the actions for the Options menu. The actions are handled by the parent widget
+        """
+        container_widget = QWidget(self)
+        layout = QHBoxLayout(container_widget)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+        container_widget.setLayout(layout)
+        self.setCornerWidget(container_widget, Qt.TopRightCorner)
+
+        run_button = QPushButton(container_widget)
+        run_button.setObjectName(self.RUN_BUTTON_OBJECT_NAME)
+        run_button.setIcon(get_icon("fa.play", color=QColor(73, 156, 84)))
```

Comment:
Maybe extract the QColor as a named class-level variable with `GREEN` in the name?